src/test/ui/type-alias-impl-trait/wf-nested-pass.rs

```diff
@@ -0,0 +1,22 @@
+// The hidden type for `impl Sized` is `IsStatic<T>`, which requires `T: 'static`.
+// We know it is well-formed because it can only be referenced as a projection:
+// <OuterOpaque<T> as Trait<&'static T>>::Out`.
+// So any instantiation of the type already requires proving `T: 'static`.
+//
+// check-pass
+
+#![feature(type_alias_impl_trait)]
+
+struct IsStatic<T: 'static>(T);
+
+trait Trait<In> {
+    type Out;
+}
+impl<T> Trait<&'static T> for () {
+    type Out = IsStatic<T>;
+}
+
+type OuterOpaque<T> = impl Trait<&'static T, Out = impl Sized>;
+fn define<T>() -> OuterOpaque<T> {}
```

Comment:
can you add a test which checks that this errors when `define` is used with a type which is not `'static`

Reply:
I've added a one in `pass_sound` revision. Note that it relies on #99217 for soundness.